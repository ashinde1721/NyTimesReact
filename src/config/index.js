export const API_URL = 'https://api.nytimes.com/svc/mostpopular/v2/mostviewed/all-sections';
export const API_KEY = 'qG808Xn6G5zAb6Wm0e6hS52JQjS6L2Er';
export const PERIOD = 7;