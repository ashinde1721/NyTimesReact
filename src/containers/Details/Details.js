import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import get from 'lodash/get';
import queryString from 'query-string';
import leftArrowIcon from '../../assets/icons/left-arrow.png'
import "./Details.scss";

class Details extends Component {

  goToArticles = () => {
    this.props.history.push(`/`);
  }

  render() {
    let queryParams = {};

    try {
      queryParams = queryString.parse(get(this.props, 'location.search'));
    } catch (err) { }

    const url = get(queryParams, 'url');
    if (!url) {
      return <div>Page Not Found</div>;
    }

    return (
      <div>
        <div className="back-btn"><img onClick={this.goToArticles.bind(this)} className="img-responsive" height="20" width="20" src={leftArrowIcon} alt="Go back" /></div>
        <iframe title='nyFrame' src={url} />
      </div>
    )
  }

}

export default withRouter(Details);