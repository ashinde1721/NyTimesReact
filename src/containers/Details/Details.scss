.back-btn{
    padding-bottom: 12px;
}

iframe{
  width: 95vw;
  height: 120vw;
}