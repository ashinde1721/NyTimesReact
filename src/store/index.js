import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducer";
import createSagaMiddleware from 'redux-saga';

import {getArticleSaga} from  '../sagas/articlesSaga'

const sagaMiddleware = createSagaMiddleware();

const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));
sagaMiddleware.run(getArticleSaga);

export default store;