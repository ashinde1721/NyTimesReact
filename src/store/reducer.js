import * as actions from '../actions/actions';

const initialState = {
    search: '',
    articlesList: []
};

const reducer = (state = initialState, action) => {
    const { type, payload = {} } = action;

    switch (type) {
        case actions.SEARCH_TEXT:
            return { ...state, search: payload.search };

        case actions.SAVE_ARTICLES:
            return {
                ...state,
                articlesList: payload.articlesList
            }
        default:
            return state;
    }
};

export default reducer;
