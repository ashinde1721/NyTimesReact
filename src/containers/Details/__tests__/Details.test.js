import React from 'react';
import { Provider } from 'react-redux';
import axios from 'axios';
import { cleanup, waitForElement, getByTitle } from 'react-testing-library';

import renderWithRouter from '../../../config/renderWithRouter';
import store from '../../../store';
import Details from '../Details';
import { API_URL, API_KEY } from '../../../config/index';

describe('Details Component', () => {
  const fetchArticles = async () => {
    const period = 7;
    const url = `${API_URL}/${period}.json?api-key=${API_KEY}`;
    const { data: { results = [] } } = await axios.get(url);
    return results;
  };

  afterEach(() => cleanup());

  it('renders component', () => {
    const renderedComponent = renderWithRouter(
      <Provider store={store}>
        <Details />
      </Provider>
    );
    expect(renderedComponent.container).toMatchSnapshot();
  });

  it('should render all NY-Times-Article-Details in the iframe for the given URL', async () => {
    const results = await fetchArticles();
    const url = results[0].url;
    const { getByTitle } = renderWithRouter(<Details />, { route: `?url=${url}` });
    
    await waitForElement(() => getByTitle('nyFrame'));
    expect(getByTitle('nyFrame').src).toEqual(url);
  },10000);
});
