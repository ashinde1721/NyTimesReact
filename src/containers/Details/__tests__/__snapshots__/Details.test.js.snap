// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component renders component 1`] = `
<div>
  <div>
    Page Not Found
  </div>
</div>
`;
